src/julia_internal.h

```diff
@@ -1220,11 +1221,19 @@ STATIC_INLINE void unbias_cong(uint64_t max, uint64_t *unbias) JL_NOTSAFEPOINT
 {
     *unbias = UINT64_MAX - ((UINT64_MAX % max) + 1);
 }
+
 STATIC_INLINE uint64_t cong(uint64_t max, uint64_t unbias, uint64_t *seed) JL_NOTSAFEPOINT
 {
-    while ((*seed = 69069 * (*seed) + 362437) > unbias)
-        ;
-    return *seed % max;
+    uint64_t mask = ~(uint64_t)0;
+    --max;
+    mask >>= __builtin_clzll(max|1);
+    uint64_t x;
+    do {
+        while ((*seed = 69069 * (*seed) + 362437) > unbias)
```

Comment:
this unbias term is now possibly biasing your results slightly and should be removed
(it is from the algorithm "Division with Rejection (Unbiased)" previously)

Reply:
Do we want to keep the api of rand_ptls and just throwaway that argument, or do I potentially just break stuff?